addons/resource/models/resource_calendar.py

```diff
@@ -368,17 +368,20 @@ def _attendance_intervals_batch(self, start_dt, end_dt, resources=None, domain=N
             res_intervals = WorkIntervals(res)
             for resource in resources:
                 if resource and resource._is_flexible():
-                    duration_days = (end - start).days + (0.5 if (end - start).total_seconds() / 3600 < resource.calendar_id.hours_per_day else 1)
-                    if resource.calendar_id and resource.calendar_id.flexible_hours:
-                        duration_hours = duration_days * resource.calendar_id.hours_per_day
-                    else:
-                        duration_hours = (end - start).total_seconds() / 3600
-                    # If the resource is flexible, return the whole period from start_dt to end_dt with a dummy attendance
-                    dummy_attendance = self.env['resource.calendar.attendance'].new({
-                        'duration_hours': duration_hours,
-                        'duration_days': duration_days,
-                    })
-                    result_per_resource_id[resource.id] = WorkIntervals([(start, end, dummy_attendance)])
+                    intervals = []
+                    start_date = start.date()
+                    end_date = end.date()
+                    num_days = (end_date - start_date).days + 1
+                    for day_offset in range(num_days):
+                        current_day = start + relativedelta(days=day_offset)
+                        day_start = current_day.replace(hour=0, minute=0, second=0)
+                        day_end = day_start + relativedelta(hours=resource.calendar_id.hours_per_day)
```

Comment:
a fully flexible resource is also taken into account in `_is_flexible` method and we should also take into account the working schedule set on the resource could be falsy. 🙂